src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/TextAreaFilter.php

```diff
@@ -0,0 +1,181 @@
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => '*' . $value . '*',
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::DOES_NOT_CONTAIN:
+                $mustNotClause = [
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => '*' . $value . '*',
+                    ],
+                ];
+
+                $filterClause = [
+                    'filter' => [
+                        'exists' => ['field' => $attributePath],
+                    ],
+                ];
+
+                $this->searchQueryBuilder
+                    ->addMustNot($mustNotClause)
+                    ->addFilter($filterClause);
+                break;
+
+            case Operators::EQUALS:
+                $attributePath .= 'raw';
+                $clause = [
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => $value,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_EQUAL:
+                $attributePath .= 'raw';
+                $MustNotClause = [
+                    'query_string' => [
+                        'default_field' => $attributePath,
+                        'query'         => $value,
+                    ],
+                ];
+
+                $filterClause = [
+                    'filter' => [
+                        'exists' => ['field' => $attributePath],
+                    ],
+                ];
+                $this->searchQueryBuilder
+                    ->addMustNot($MustNotClause)
+                    ->addFilter($filterClause);
+                break;
+
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            default:
+                throw new \InvalidArgumentException(sprintf('This filter does not support operator "%s".', $operator));
+        }
+
+        return $this;
+    }
+
+    /**
+     * Escapes value
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    protected function escapeValue($value)
+    {
+        $regex = '#[-+=|! &(){}\[\]^"~*<>?:/\\\]#';
```

Comment:
Yep :/ https://github.com/akeneo/ElasticSearchBundle/blob/master/Query/Filter/StringFilter.php#L213

I was no intention of leaving this as is. More to have your opinion on the "escaping" the value before research.